source/admin_manual/list_of_events.rst

```diff
@@ -2726,6 +2726,91 @@ Event emitted when dict server finishes transaction. Same fields as
++---------------------+------------------------------------------------------+
+| remote_ip           | Remote IP address                                    |
++---------------------+------------------------------------------------------+
+| remote_port         | Remote port                                          |
++---------------------+------------------------------------------------------+
+| user                | Full username                                        |
++---------------------+------------------------------------------------------+
+
+Proxy Login Common Fields
+-------------------------
+
++---------------------+------------------------------------------------------+
+| Field               | Description                                          |
++=====================+======================================================+
+| Inherits from :ref:`pre_login_client`                                      |
++---------------------+------------------------------------------------------+
+| service             | Name of service e.g. ``submission``, ``imap``        |
++---------------------+------------------------------------------------------+
+| dest_host           | Host name of the proxy remote (if proxying is        |
+|                     | configured with IP address, will have the same value |
+|                     | as dest_ip).                                         |
++---------------------+------------------------------------------------------+
+| dest_ip             | Destination IP for proxying                          |
++---------------------+------------------------------------------------------+
+| dest_port           | Destination port for proxying                        |
++---------------------+------------------------------------------------------+
+| source_ip           | Source IP where proxying originated from             |
++---------------------+------------------------------------------------------+
+
+login_proxy_started
+^^^^^^^^^^^^^^^^^^^
+Emitted before connecting to proxy remote.
+
+login_proxy_logged_in
+^^^^^^^^^^^^^^^^^^^^^
+Emitted after proxied connection is established and user is successfully logged
+in to the backend.
+
++---------------------+------------------------------------------------------+
+| Field               | Description                                          |
++=====================+======================================================+
+| source_port         | Source port where proxying originated from (present  |
+|                     | after proxy connection is established                |
++---------------------+------------------------------------------------------+
+| master_user         | If proxying is done with a master user               |
+|                     | authentication, contains the full username of master |
+|                     | user.                                                |
```

Comment:
Isn't this (or couldn't it be) already in the _started event as well, so it could be in the common fields?

Reply:
Not with current code. `client->proxy_master_user` is filled after login_proxy_new() where _started event is emitted.

Reply:
Updated code & here.

Reply:
Forgot to remove master_user here.